src/main/java/hudson/plugins/analysis/core/BuildHistory.java

```diff
@@ -330,7 +333,7 @@ public AbstractHealthDescriptor getHealthDescriptor() {
      * @deprecated use {@link #BuildHistory(AbstractBuild, Class, boolean, boolean)}
      */
     @Deprecated
-    public BuildHistory(final AbstractBuild<?, ?> baseline, final Class<? extends ResultAction<? extends BuildResult>> type,
+    public BuildHistory(final Run<?, ?> baseline, final Class<? extends ResultAction<? extends BuildResult>> type,
```

Comment:
Why are you changing the signature of this constructor? It was already deprecated.